agenta-cli/agenta/sdk/decorators/tracing.py

```diff
@@ -58,19 +58,20 @@ async def async_wrapper(*args, **kwargs):
             try:
                 result = await func(*args, **kwargs)
                 self.tracing.update_span_status(span=span, value="OK")
-            except Exception as e:
-                result = str(e)
-                self.tracing.set_span_attribute(
-                    {"traceback_exception": traceback.format_exc()}
-                )
-                self.tracing.update_span_status(span=span, value="ERROR")
-            finally:
                 self.tracing.end_span(
                     outputs=(
                         {"message": result} if not isinstance(result, dict) else result
                     )
                 )
-            return result
+                return result
+
+            except Exception as e:
+                self.tracing.set_span_attribute(
+                    {"traceback_exception": traceback.format_exc()}
+                )
+                self.tracing.update_span_status(span=span, value="ERROR")
+                self.tracing.end_span(outputs={})
```

Comment:
Currently, we are not displaying the `traceback_exception` in the traces/generations view on the frontend, so it makes that we include the error message in the span output so that the user can understand what happened. 

```suggestion
                self.tracing.end_span(outputs={"message": str(e)})
```

Reply:
But how would this message be shown. It should somehow be shown in a special color way in the FE. 
I think it might make more sense to return the error message in the form of meta data not as output, since it is semantically different. Maybe @aybruhm or @jp-agenta can look in how it is usually done in otel?

Reply:
There are semantic conventions for exceptions on spans: https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/

Reply:
> But how would this message be shown. It should somehow be shown in a special color way in the FE.

The FE would show the error (as a message) in red if the span status is "ERROR".

> There are semantic conventions for exceptions on spans: https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/

Thank you for looking into it, @jp-agenta.  

> I think it might make more sense to return the error message in the form of meta data not as output, since it is semantically different.

@mmabrouk, yup, this makes sense. 

So, something like this? As documented in the link @jp-agenta atttached?

```python
result = {
    "message": str(e),
    "stacktrace": traceback.format_exc(),
}
self.tracing.set_span_attribute(
    {"exception": result}
)
```

Reply:
We would need `self.tracing.add_event` and then a more specialized version of `add_event` called `self.tracing.record_exception`. This would require a bit of work. Should we include that work in this issue or should I create a new one ? @mmabrouk

Reply:
hmm... We don't have currently events in our SDK.
I suggest therefore:
- Opening an issue for correctly handling exceptions in observability (both from the SDK and the UI) and setting the priority to high. We might decide to handle these using meta-data or a simpler solution that events at this point of time. But the FE work is non-trivial in both cases and out of scope for this PR. (can you please do that @jp-agenta )
- Testing Abram's solution (does it render correctly and without bugs in the UI). If it does, let's keep it, otherwise let's revert to `outputs={}` and not work on it

Reply:
refs:
1. https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/trace/api.md
2. https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/trace/exceptions.md
3. https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/
4. https://opentelemetry.io/docs/concepts/signals/traces/

Reply:
On it.

Reply:
**API key error** 👇 
<img width="1386" alt="Screenshot 2024-07-05 at 14 49 29" src="https://github.com/Agenta-AI/agenta/assets/174311389/ca0e3b2d-208e-4870-992d-8acbfe84bce8">
**Third-party auth error** 👇 
<img width="1383" alt="Screenshot 2024-07-05 at 14 52 03" src="https://github.com/Agenta-AI/agenta/assets/174311389/00cb8f6e-da8e-4b7e-a998-efc47e62d966">
**no errors** 👇 
<img width="1392" alt="Screenshot 2024-07-05 at 14 53 25" src="https://github.com/Agenta-AI/agenta/assets/174311389/138abf03-5167-4dd6-9400-3526793cab76">

This is with 
`result = { "message": str(e), "stacktrace": traceback.format_exc()}`
and
`self.tracing.end_span(outputs=result)`

Reply:
Will create the issue.